libs/blocks/global-navigation/global-navigation.js

```diff
@@ -373,6 +431,81 @@ class Gnav {
     decorationTimeout = setTimeout(decorateDropdown, CONFIG.delays.loadDelayed);
   };
 
+  decorateUniversalNav = async () => {
+    const config = getConfig();
+    let language;
+    let region;
+    if (config.locale.ietf.includes('-')) {
+      [language, region] = config.locale.ietf.split('-');
+    } else {
+      [region, language] = config.locale.prefix.replace('/', '').split('_');
+    }
+    const locale = `${language.toLowerCase()}_${region.toUpperCase()}`;
+    const environment = config.env.name === 'prod' ? 'prod' : 'dev';
+    const visitorGuid = window.alloy ? await window.alloy('getIdentity').then((data) => data?.identity?.ECID) : undefined;
+    const getDevice = () => {
+      const agent = navigator.userAgent;
+      if (agent.includes('Mac')) return 'macOS';
+      if (agent.includes('Win')) return 'windows';
+      if (agent.includes('Linux')) return 'linux';
+      if (agent.includes('CrOS')) return 'chromeOS';
+      if (agent.includes('Android')) return 'android';
+      if (agent.includes('iPad')) return 'iPadOS';
+      if (agent.includes('iPhone')) return 'iOS';
+
+      return 'linux';
+    };
+
+    await Promise.all([
+      loadScript(`https://${environment}.adobeccstatic.com/unav/1.0/UniversalNav.js`),
+      loadStyle(`https://${environment}.adobeccstatic.com/unav/1.0/UniversalNav.css`),
+    ]);
+
+    const getChildren = () => {
+      const children = [CONFIG.universalNav.components.profile];
+      // reset sign up value on change
+      children[0].attributes.isSignUpRequired = false;
+
+      if (isDesktop.matches) {
+        this.universalNavComponents?.forEach((component) => {
+          if (component === 'profile') return;
+          if (component === 'signup') {
+            children[0].attributes.isSignUpRequired = true;
+            return;
+          }
+
+          children.push(CONFIG.universalNav.components[component]);
+        });
+      }
+
+      return children;
+    };
+
+    const getConfiguration = () => ({
+      target: this.blocks.universalNav,
+      env: environment,
+      locale,
+      imsClientId: window.adobeid?.client_id,
+      analyticsContext: {
+        consumer: {
+          name: 'adobecom',
+          version: '1.0',
+          platform: 'Web',
+          device: getDevice(),
+          os_version: navigator.platform,
+        },
+        event: { visitor_guid: visitorGuid },
+      },
+      children: getChildren(),
+    });
+
+    window.UniversalNav(getConfiguration());
```

Comment:
Are we absolutely positive that `window.UniversalNav` will exist? Should we add some optional chaining, just in case?

Reply:
yes

Reply:
Even if the scripts don't load due to network issues? Do we stop execution early in that case?